test/DynamoCoreWpfTests/ViewExtensions/WorkspaceDependencyViewExtensionTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Windows;
+using Dynamo.Configuration;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Scheduler;
+using Dynamo.WorkspaceDependency;
+using Dynamo.Wpf.Extensions;
+using NUnit.Framework;
+
+namespace DynamoCoreWpfTests
+{
+    public class WorkspaceDependencyViewExtensionTests : DynamoTestUIBase
+    {
+        private WorkspaceDependencyViewExtension viewExtension = new WorkspaceDependencyViewExtension();
+
+        private string PackagesDirectory { get { return Path.Combine(GetTestDirectory(this.ExecutingDirectory), "pkgs"); } }
+
+        protected override DynamoModel.IStartConfiguration CreateStartConfiguration(IPathResolver pathResolver)
+        {
+            return new DynamoModel.DefaultStartConfiguration()
+            {
+                PathResolver = pathResolver,
+                StartInTestMode = true,
+                GeometryFactoryPath = preloader.GeometryFactoryPath,
+                ProcessMode = TaskProcessMode.Synchronous,
+                Preferences = new PreferenceSettings() { CustomPackageFolders = new List<string>() { this.PackagesDirectory } }
+            };
+        }
+
+        [Test]
+        public void KeepInstalledVersionOfPackageTest()
+        {
+            RaiseLoadedEvent(this.View);
+            var extensionManager = View.viewExtensionManager;
+            extensionManager.Add(viewExtension);
+
+            var loadedParams = new ViewLoadedParams(View, ViewModel);
+            viewExtension.DependencyView = new WorkspaceDependencyView(viewExtension, loadedParams);
+
+            System.Version infoVersion = new System.Version("0.1.3");
+            PackageDependencyInfo info = new PackageDependencyInfo("GetHighest", infoVersion);
+
+            // Once choosing to keep the installed loaded version, info.Version should reflect the lower version
+            viewExtension.DependencyView.UpdateWorkspaceToUseInstalledPackage(info);
+            Assert.AreEqual("0.1.2", info.Version.ToString());
+        }
+
+        [Test]
+        public void WillNotAddDupWorkspaceDependencyViewExtensionLoadTest()
+        {
+            RaiseLoadedEvent(this.View);
+
+            var extensionManager = View.viewExtensionManager;
+
+            var initialNum = View.TabItems.Count;
+
+            // Adding the workspace dependency view extension will 
+            // not add a dup tab in the extensions side bar
+            extensionManager.Add(viewExtension);
```

Comment:
Why will there be a dup in this case and why is it not incremented? I guess I do not understand this test.

Reply:
@aparajit-pratap The extension is already loaded when Dynamo is launched but the view will not show unless you open a graph, so calling the function again will not inject the extension tab because it is already there